docs/spec.md

```diff
@@ -4361,20 +4373,36 @@ ordering of `index_space(result)` and `index_space(operand)`.
 
 #### Inputs
 
-| Label | Name      | Type                                  | Constraints |
-|-------|-----------|---------------------------------------|-------------|
-| (I1)  | `operand` | tensor or per-tensor quantized tensor | (C1-C2)     |
+| Label | Name      | Type                       | Constraints |
+|-------|-----------|----------------------------|-------------|
+| (I1)  | `operand` | tensor or quantized tensor | (C1-C3)     |
 
 #### Outputs
 
-| Name     | Type                                  | Constraints |
-|----------|---------------------------------------|-------------|
-| `result` | tensor or per-tensor quantized tensor | (C1-C2)     |
+| Name     | Type                       | Constraints |
+|----------|----------------------------|-------------|
+| `result` | tensor or quantized tensor | (C1-C3)     |
 
 #### Constraints
 
-* (C1) `element_type(operand) = element_type(result)`.
+* (C1) `element_type(result)` is given by:
+  * `element_type(operand)`, if `!is_per_axis_quantized(operand)`.
+  * `element_type(operand)` except that `quantization_dimension(operand)` and
+    `quantization_dimension(result)` may differ, otherwise.
 * (C2) `size(operand) = size(result)`.
+* (C3) If `is_per_axis_quantized(operand)`:
+  * `reduce(dims(operand, [0, 1, ..., quantization_dimension(operand) - 1]),
```

Comment:
Do we use this `reduce` notation, or something similar, elsewhere in the spec? I like that this will match the implementation fairly closely, but was a bit more difficult to understand at first glance.

I think this concern will be resolved once we have a verifier and an error message. I anticipate this error message will be a little difficult to write.. "expected dimensions of rank < quantizedDim of input and output to have the same number of elements" ?

Reply:
Yes this notation is used in the following ops. 
[batch_norm_grad](https://github.com/openxla/stablehlo/blob/main/docs/spec.md#batch_norm_grad)
[batch_norm_training](https://github.com/openxla/stablehlo/blob/main/docs/spec.md#batch_norm_training)
[dot_general](https://github.com/openxla/stablehlo/blob/main/docs/spec.md#dot_general)
[reduce_window](https://github.com/openxla/stablehlo/blob/main/docs/spec.md#reduce_window)
[select_and_scatter](https://github.com/openxla/stablehlo/blob/main/docs/spec.md#select_and_scatter)

Or maybe something like `quantization_dimension(operand) cannot be splitted or merged during the reshape operation`